### PR TITLE
[Validator] synchronize IBAN formats

### LINKED ISSUE #58126: [Validator] Some countries are missing
### Symfony version(s) affected

6.4.10

### Description

Some countries are not included and IbanValidator warns.

### How to reproduce

Try to submit this iban `MN140005005163847716`.

```php
<?php
namespace App\Form;

use Symfony\Component\Form\AbstractType;
use Symfony\Component\Form\Extension\Core\Type\SubmitType;
use Symfony\Component\Form\Extension\Core\Type\TextType;
use Symfony\Component\Form\FormBuilderInterface;
use Symfony\Component\Validator\Constraints as Assert;

class IbanValidatorFormType extends AbstractType
{
    public function buildForm(FormBuilderInterface $builder, array $options): void
    {
        $builder
            ->add('iban', TextType::class, [
                'label' => 'iban',
                'constraints' => [
                    new Assert\Iban()
                ],
                'attr' => [
                    'placeholder' => 'enter_your_iban_here'
                ]
            ])
            ->add('submit', SubmitType::class, [
                'label' => 'validate'
            ])
        ;
    }
}

```

### Possible Solution

I see there is an update script `Resources/bin/sync-iban-formats.php`. The list will possibly updated when the script executed.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | Fix #58126
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/Validator/Tests/Constraints/IbanValidatorTest.php [hunk 1]
```diff
@@ -109,6 +109,7 @@ public static function getValidIbans()
             ['MU17 BOMM 0101 1010 3030 0200 000M UR'], // Mauritius
             ['MD24 AG00 0225 1000 1310 4168'], // Moldova
             ['MC93 2005 2222 1001 1223 3M44 555'], // Monaco
+            ['MN14 0005 0051 6384 7716'], // Mongolia
```
Comment (alexandre-daubois) [inline src/Symfony/Component/Validator/Tests/Constraints/IbanValidatorTest.php:113]: Should other IBAN formats be added to the data provider?

**Review by alexandre-daubois — COMMENTED** [review]
Serial fixer :rocket:

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/Validator/Tests/Constraints/IbanValidatorTest.php (hunk 1 above)
Comment (xabbuh) [reply src/Symfony/Component/Validator/Tests/Constraints/IbanValidatorTest.php:113]: I see if I can find example IBANs for the other added countries

Comment (tugrul) [reply src/Symfony/Component/Validator/Tests/Constraints/IbanValidatorTest.php:113]:
```
FK12SC987654321098
OM040280000012345678901
YE09CBKU0000000000001234560101
```

**Review by stof — APPROVED** [review]

File: src/Symfony/Component/Validator/Tests/Constraints/IbanValidatorTest.php (hunk 1 above)
Comment (xabbuh) [reply src/Symfony/Component/Validator/Tests/Constraints/IbanValidatorTest.php:113]: added, thanks 👍

**Review by alexandre-daubois — APPROVED** [review]

**Review by derrabus — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]

**Review by tugrul — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**